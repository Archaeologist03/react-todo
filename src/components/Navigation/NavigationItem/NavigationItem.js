import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { logoutUser } from '../../../store/actions/authActions';

import './NavigationItem.scss';

function NavigationItem(props) {
  const openingModal = () => {
    // if has modal(login/singup) open it, otherwise(logout) null
    if (props.openModal) {
      props.openModal(true);
    } else {
      return null;
    }
  };

  const loggingOut = () => {
    if (props.loggingOut) {
      props.onLogoutUser();
      props.history.push('/');
    }
  };

  return (
    <NavLink
      onClick={() => {
        openingModal();
        loggingOut();
      }}
      className='nav-item'
      activeClassName='active-nav-item'
      to={props.linkTo}>
      <span className='nav-text-span'>{props.children}</span>
    </NavLink>
  );
}

const mapStateToProps = state => {
  return {
    isLogged: state.app.isLogged,
  };
};
const mapDispatchToProps = dispatch => {
  return {
    onLogoutUser: () => dispatch(logoutUser()),
  };
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(NavigationItem),
);
