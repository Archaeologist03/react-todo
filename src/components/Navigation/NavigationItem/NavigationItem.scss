.nav-item {
  border: 2px solid rgb(243, 87, 31);
  border-radius: 10%;
  padding: 5px;
  font-size: 1.4rem;
  text-decoration: none;
  color: orangered;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  transition: .7s;
  margin: 0 10px;

  

  &:hover {
    background-color: rgba(231, 91, 40, 0.89);
    color: rgba(245, 245, 245, 0.925);
    
  }
}

