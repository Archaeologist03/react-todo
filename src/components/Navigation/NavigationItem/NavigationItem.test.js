import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { NavigationItem } from './NavigationItem';

describe('NavigationItem', () => {
  it('renders NavigationItem without crashing given the required props', () => {
    const props = {
      isLogged: false,
      onLogoutUser: jest.fn(),
      linkTo: '/login',
    };
    const wrapper = shallow(<NavigationItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call openingModal fn when clicked', () => {
    const props = {
      isLogged: false,
      onLogoutUser: jest.fn(),
      linkTo: '/login',
      children: 'Login',
    };

    const loggingOut = () => {
      props.onLogoutUser();
    };

    const wrapper = shallow(<NavigationItem {...props} />);
    const navItemClassElement = wrapper.find('.nav-item');

    navItemClassElement.simulate('click', loggingOut());
    expect(props.onLogoutUser).toBeCalled();
  });
});
