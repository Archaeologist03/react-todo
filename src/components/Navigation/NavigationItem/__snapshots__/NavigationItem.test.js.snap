// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem renders NavigationItem without crashing given the required props 1`] = `
<NavLink
  activeClassName="active-nav-item"
  className="nav-item"
  onClick={[Function]}
  to="/login"
>
  <span
    className="nav-text-span"
  />
</NavLink>
`;
